cmake/linux/package_linux.sh.in

```diff
@@ -62,14 +66,20 @@ echo -e "\nDownloading linuxdeployqt to ${LINUXDEPLOYQT}..."
 if env -i which linuxdeployqt > /dev/null 2>&1; then
 	skipped "System already provides this utility"
 else
-	filename="linuxdeployqt-continuous-$(uname -p).AppImage"
+	filename="linuxdeployqt-continuous-$ARCH.AppImage"
 	url="https://github.com/probonopd/linuxdeployqt/releases/download/continuous/$filename"
 	down_file="$(pwd)/$filename"
-	if [ ! -f "$LINUXDEPLOYQT" ]; then
+	if [ ! -L "$LINUXDEPLOYQT" ]; then
```

Comment:
`-f` tests for regular files and `-L` tests for symbolic links. If we just want to test existence, neither looks like a good candidate.

Reply:
Prior to this, if you deleted the downloaded file and tried to build again, the `-f` test would be false (broken link) and it would try to create symlink again which would fail.

I've blindly assumed no one is manually putting things in the `build` directory, but we could test if `$LINUXDEPLOYQT` exists as a regular file and then skip the download.

In that case, it would never check for a newer version of it.

> When running Wget with -N, the decision as to whether or not to download a newer copy of a file depends on the local and remote timestamp and size of the file.

But I don't know how relevant that is... What do you think?

Reply:
> Prior to this, if you deleted the downloaded file and tried to build again, the -f test would be false (broken link) and it would try to create symlink again which would fail.

How about adding `-f`(force) flag to `ln`?

> I've blindly assumed no one is manually putting things in the `build` directory

Usually true, unless they are on a machine with limited internet connectivity.

BTW, @tresf, do you have any ideas?

Reply:
I believe [`-e`](https://stackoverflow.com/a/40082454/3196753) is what you're looking for if you don't care what type of file it is.  I'm not sure if this is POSIX compliant, but testing on MacOS's `sh`, `zsh` and their old `bash` all worked fine.

Reply:
> https://github.com/LMMS/lmms/commit/418bcce40262a57447f0e1786d5bb86feef8f410

The symbolic link was added by you @PhysSong here https://github.com/LMMS/lmms/commit/418bcce40262a57447f0e1786d5bb86feef8f410. 😄.  If the script expects a symbolic link, `-L` seems warranted, no?

Reply:
> unless they are on a machine with limited internet connectivity.

This is always a good consideration to have.  If that's the case, perhaps stop asking wget how old the file is? 😄Partially kidding... but controlling the re-download logic is always a good idea for metered connections.

Reply:
> > BTW, @tresf, do you have any ideas?
> 
> The symbolic link was added by you @PhysSong here [418bcce](https://github.com/LMMS/lmms/commit/418bcce40262a57447f0e1786d5bb86feef8f410). smile. If the script expects a symbolic link, `-L` seems warranted, no?

Oh, it's not me.

Reply:
> Oh, it's not me.

Sorry!

Reply:
If I understand

1. Check for existing env variable LINUXDEPLOYQT 
2. Use as-if if it's a regular file
3. Download latest version if it's a symlink
4. Create symlink if it does not exist

Reply:
^ I like this logic. :)

Reply:
I went with

1. Use value of `LINUXDEPLOYQT`
2. Use already downloaded version
3. Use `linuxdeployqt` from PATH
4. Download latest version

Reply:
So if the executable is already downloaded and unpacked it won't try to download the latest version... You could call this a regression but I think it's forgivable/expected.

Reply:
> 

Do you mean manually?  If so, I agree, this is consistent with how people with no internet/metered connections would expect it to behave.

Reply:
Eh, I didn't mean manually. I realized my mistake while typing that comment but I was too tired to fix it. But now I have fixed it so it downloads the latest version every time *unless* you have specified `LINUXDEPLOYQT` and `APPIMAGETOOL`, or they exist in PATH.

Reply:
Sounds good to me.  @PhysSong ?

Reply:
No objections to the idea.